test/unit/functions/validation/invalidFunctionAssignments.spec.ts

```diff
@@ -1,191 +1,153 @@
-    UnsupportedSelfFunctionConversion,
-} from "../../../../src/transformation/utils/errors";
+    unsupportedOverloadAssignment,
+    unsupportedNoSelfFunctionConversion,
+    unsupportedSelfFunctionConversion,
+} from "../../../../src/transformation/utils/diagnostics";
 import * as util from "../../../util";
 import { invalidTestFunctionAssignments, invalidTestFunctionCasts } from "./functionPermutations";
 
 test.each(invalidTestFunctionAssignments)(
     "Invalid function variable declaration (%p)",
     (testFunction, functionType, isSelfConversion) => {
-        const code = `
+        util.testModule`
             ${testFunction.definition || ""}
             const fn: ${functionType} = ${testFunction.value};
-        `;
-        const err = isSelfConversion
-            ? UnsupportedSelfFunctionConversion(util.nodeStub)
-            : UnsupportedNoSelfFunctionConversion(util.nodeStub);
-        expect(() => util.transpileString(code, undefined, false)).toThrowExactError(err);
+        `.expectDiagnosticsToMatchSnapshot(
+            [isSelfConversion ? unsupportedSelfFunctionConversion.code : unsupportedNoSelfFunctionConversion.code],
+            true
+        );
     }
 );
 
 test.each(invalidTestFunctionAssignments)(
     "Invalid function assignment (%p)",
     (testFunction, functionType, isSelfConversion) => {
-        const code = `
+        util.testModule`
             ${testFunction.definition || ""}
             let fn: ${functionType};
             fn = ${testFunction.value};
-        `;
-        const err = isSelfConversion
-            ? UnsupportedSelfFunctionConversion(util.nodeStub)
-            : UnsupportedNoSelfFunctionConversion(util.nodeStub);
-        expect(() => util.transpileString(code, undefined, false)).toThrowExactError(err);
+        `.expectDiagnosticsToMatchSnapshot(
+            [isSelfConversion ? unsupportedSelfFunctionConversion.code : unsupportedNoSelfFunctionConversion.code],
+            true
+        );
     }
 );
 
-test.each(invalidTestFunctionCasts)(
-    "Invalid function assignment with cast (%p)",
-    (testFunction, castedFunction, isSelfConversion) => {
-        const code = `
-            ${testFunction.definition || ""}
-            let fn: typeof ${testFunction.value};
-            fn = ${castedFunction};
-        `;
-        const err = isSelfConversion
-            ? UnsupportedSelfFunctionConversion(util.nodeStub)
-            : UnsupportedNoSelfFunctionConversion(util.nodeStub);
-        expect(() => util.transpileString(code, undefined, false)).toThrowExactError(err);
-    }
-);
+test.each(invalidTestFunctionCasts)("Invalid function assignment with cast (%p)", (testFunction, castedFunction) => {
+    util.testModule`
+        ${testFunction.definition || ""}
+        let fn: typeof ${testFunction.value};
+        fn = ${castedFunction};
+    `.expectDiagnosticsToMatchSnapshot(
+        [unsupportedNoSelfFunctionConversion.code, unsupportedSelfFunctionConversion.code],
```

Comment:
You seem to be missing the `isSelfConversion` that the other tests have. Are these really both thrown here? If not, why is this test not failing?

Reply:
Oh is this because of line 87?

Reply:
`with cast` tests are a bit different from others, unlike other tests they have 2 different errors - one in the cast and another in the assignment. Besides that, in 2 of these  tests we don't care about the order (seemingly because diagnostics are sorted by code spans), but in `Invalid function return with cast` it does